File: app/src/main/java/com/ashlikun/baseproject/core/base/iview/IBaseSwipeView.java
package com.ashlikun.baseproject.core.base.iview;

import com.ashlikun.core.iview.BaseView;
import com.ashlikun.xrecycleview.RefreshLayout;

/**
 * Created by Administrator on 2016/8/8.
 * <p>
 * 下拉刷新的基类接口
 */

public interface IBaseSwipeView extends BaseView, RefreshLayout.OnRefreshListener {
    /**
     * 获取下拉刷新控件
     */
    RefreshLayout getSwipeRefreshLayout();
}

File: libcore/src/main/java/com/ashlikun/libcore/mvp/iview/IBaseSwipeView.kt
package com.ashlikun.libcore.mvp.iview

import com.ashlikun.core.iview.BaseView
import com.ashlikun.xrecycleview.RefreshLayout

/**
 * Created by Administrator on 2016/8/8.
 *
 *
 * 下拉刷新的基类接口
 */

interface IBaseSwipeView : BaseView, RefreshLayout.OnRefreshListener {

    /**
     * 获取下拉刷新控件
     */
    fun getSwipeRefreshLayout(): RefreshLayout
}
